### DISCUSSION TITLE
Ajax Cart for Dawn theme

### DISCUSSION THREAD

**Discussion by AsadBaig-ProblemSolver** [EvgeniyMukhamedjanov/liquid-ajax-cart#4] [Q&A]
I have applied the same code everything is good but when i applied "data-ajax-cart-toggle-class-button" it didn't show me the floating cart drawer. Could you please help me regarding this?

Following is the code for cart-icon in header i have applied the toggle class button to anchor tag
`<a href="{{ routes.cart_url }}" class="header__icon header__icon--cart link focus-inset"  data-ajax-cart-toggle-class-button="js-my-cart-open"  id="cart-icon-bubble" style=" position: absolute; /* color: white; */ left: 76%; top: 15%;" >
        {%- liquid
          if cart == empty
            render 'icon-cart-empty'
          else
            render 'icon-cart'
          endif
        -%}
        <span class="visually-hidden">{{ 'templates.cart.cart' | t }}</span>
        {%- if cart != empty -%}
          <div class="cart-count-bubble">
            {%- if cart.item_count < 100 -%}
              <span aria-hidden="true"  data-ajax-cart-bind-state="cart.item_count">{{ cart.item_count }}</span>
            {%- endif -%}
            <span class="visually-hidden">{{ 'sections.header.cart_count' | t: count: cart.item_count }}</span>
          </div>
        {%- endif -%}
      </a>`

The code for theme.liquid:
`<div class="my-container">
      <div class="my-container__content">
     {% section 'Slider-Header' %}
    <main id="MainContent" class="content-for-layout focus-none" role="main" tabindex="-1">
      {{ content_for_layout }}
    </main>
    {% section 'Slider-Footer' %}
      </div>
      <div class="my-container__cart">
        {% section 'my-cart' %}
      </div>
    </div>`

The CSS Code for styling:
`.my-container{
  display: flex;
}

.my-container__content{
    flex: 1 1 auto;

}
.my-container__cart{
    display:none;
}
body.js-my-cart-open.my-container__cart{
     flex: 0 0 300px;
     padding: 15px;
}

body.js-ajax-cart-request-in-progress .my-cart__items{
  opacity: .5;

}
form [action="cart/add"].js-ajax-cart-form-in-progress [type="submit"]{
  opacity:.5;
}
body.js-ajax-cart-empty .cart-count-bubble{
   display: none !important;
}
body.js-ajax-cart-not-empty .cart-count-bubble{
   display: inline !important;
}
`

I'm using the dawn theme.
Thank you

**Comment by EvgeniyMukhamedjanov** [comment]
Hi @AsadBaig-ProblemSolver.
Could you please share a preview link of your theme or password of the store if it is protected. I'll check what is the problem.

**Comment by AsadBaig-ProblemSolver** [comment]
Yeah Sure,
https://sliders-world.myshopify.com/products/nike-shoes
Password: xangam
Please let me know when you get to link

**Comment by EvgeniyMukhamedjanov** [comment]
I believe you missed a space between `body.js-my-cart-open` and  `.my-container__cart` in the CSS. It should be like this:
```css
body.js-my-cart-open .my-container__cart{ 
```

@AsadBaig-ProblemSolver, let me know if it helps

**Reply by AsadBaig-ProblemSolver** [reply]
Okay Let me implement this

**Reply by AsadBaig-ProblemSolver** [reply]
No it didn't work. I have implement this

**Comment by EvgeniyMukhamedjanov** [comment]
@AsadBaig-ProblemSolver 
Also can you add `display: block` within:
```css
body.js-my-cart-open .my-container__cart { 
    flex: 0 0 300px;
    padding: 15px;
    display: block;
}
```

**Reply by AsadBaig-ProblemSolver** [reply]
I have use display: block but didn't get results

I have another question 
Could i show the drawer on clicking the "Add to cart" button?

**Reply by EvgeniyMukhamedjanov** [reply]
@AsadBaig-ProblemSolver 
I checked your shop — the cart drawer appears and disappears on a cart button click, but it is out of screen area. Try to scroll right after you click the cart button — you'll see the cart drawer. Pay attention that the horizontal scroll appears when you click the cart button. So you need to review the CSS of the page layout to fix it.

In order to show the drawer after adding to cart you need to replace this code:
```liquid
<script type="module">
  import '{{ "liquid-ajax-cart-v1.6.0.js" | asset_url }}';
</script>
```
with the folowing:
```liquid
<script type="module">
  import { subscribeToCartAjaxRequests } from '{{ "liquid-ajax-cart-v1.6.0.js" | asset_url }}';
  subscribeToCartAjaxRequests(( requestState, subscribeToResult ) => {
	subscribeToResult( requestState => {
		if ( requestState.requestType === 'add' && requestState.responseData?.ok ) {
			document.body.classList.add('js-my-cart-open');
		}
	})
  })
</script>
```

**Reply by AsadBaig-ProblemSolver** [reply]
Okay so i have to assign function to cart button on toogle?

**Reply by EvgeniyMukhamedjanov** [reply]
@AsadBaig-ProblemSolver 
The js-function that I shared is intended to add the `js-my-cart-open` to the `body` tag after successful adding to cart.

The cart button (data-ajax-cart-toggle-class-button="js-my-cart-open") adds the `js-my-cart-open`  to the `body` tag when you click on it (because of data-ajax-cart-toggle-class-button attribute).

You need to make sure that the cart drawer becomes visible when the .js-my-cart-open is attached to the `body` tag.

**Comment by AsadBaig-ProblemSolver** [comment]
Hello EvgeniyMukhamedjanov/liquid-ajax-cart
I have made some changes and please have a look on it. But I have an issues
when cart drawer appear there's an issue in styling of the page. All other
things become squeezed.
Also how could I add discount code option to the cart drawer?

**Reply by AsadBaig-ProblemSolver** [reply]
Also i want  a clouser button in my cart so when i clicked on it cart will be closed

**Reply by EvgeniyMukhamedjanov** [reply]
Hi @AsadBaig-ProblemSolver 
Discount codes are applied only on checkout page as far as I know. Or you need to find an app.
For a close button you can use `data-ajax-cart-toggle-class-button="js-my-cart-open | remove"` attribute.
Regarding styles — Liquid Ajax Cart doesn't have any influence on styling, so CSS depends only on you. You can change it as you want.

**Comment by AsadBaig-ProblemSolver** [comment]
Yes I have done with close button.
But I have issue with cart position. When cart opens it changes the whole
website position.
Could you please checked it.
I have place my cart section in theme.liquid file.
That's why  I think there's issue with the styling. Can place my cart some
other way so it didn't affect my other theme styling.

**Comment by AsadBaig-ProblemSolver** [comment]
`
    <div class="my-container">
      <div class="my-container__content" style="overflow-x:clip; margin:0; padding: 0;">
     {% section 'Slider-Header' %}
    <main id="MainContent" class="content-for-layout focus-none" role="main" tabindex="-1">
      {{ content_for_layout }}
    </main>
    {% section 'Slider-Footer' %}
      </div>
      <div class="my-container__cart" style="margin:0; padding: 0;">
        {% section 'my-cart' %}
      </div>
    </div>`

That's the code that i have used in theme.liquid file.
Is there any way to put my ajax-cart other than in theme file

**Reply by EvgeniyMukhamedjanov** [reply]
The ajax-cart is inside `{% section 'my-cart' %}`. You need to include it to the place where you want to show it.

**Reply by AsadBaig-ProblemSolver** [reply]
Can i take this section to product-template or in place of cart page section?

**Reply by EvgeniyMukhamedjanov** [reply]
You can place it within product template, but you can't put it inside another section because the 'my-cart' is section itself

**Reply by AsadBaig-ProblemSolver** [reply]
OKAY I WILL NOT TRY IT
